src/bci_build/package.py

```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
+
+# apache fails to start without its log folder
+RUN mkdir -p /var/log/apache2
+
+WORKDIR /srv/www/htdocs
+
+EXPOSE 80
+"""
+        )
+    elif php_variant == PhpVariant.fpm:
+        extra_pkgs = [f"apache2-mod_php{php_version}"]
+        extra_env = {}
+        cmd = ["php-fpm"]
+        custom_end = r"""WORKDIR /srv/www/htdocs
+
+RUN set -eux; \
+	cd /etc/php8/fpm/; \
+        cp php-fpm.d/www.conf.default php-fpm.d/www.conf; \
+        cp php-fpm.conf.default php-fpm.conf; \
+	{ \
+		echo '[global]'; \
+		echo 'error_log = /proc/self/fd/2'; \
+		echo; echo '; https://github.com/docker-library/php/pull/725#issuecomment-443540114'; echo 'log_limit = 8192'; \
+		echo; \
+		echo '[www]'; \
+		echo '; if we send this to /proc/self/fd/1, it never appears'; \
+		echo 'access.log = /proc/self/fd/2'; \
+		echo; \
+		echo 'clear_env = no'; \
+		echo; \
+		echo '; Ensure worker stdout and stderr are sent to the main error log.'; \
+		echo 'catch_workers_output = yes'; \
+		echo 'decorate_workers_output = no'; \
+	} | tee php-fpm.d/docker.conf; \
+	{ \
+		echo '[global]'; \
+		echo 'daemonize = no'; \
+	} | tee php-fpm.d/zz-docker.conf
+
+# Override stop signal to stop process gracefully
+# https://github.com/php/php-src/blob/17baa87faddc2550def3ae7314236826bc1b1398/sapi/fpm/php-fpm.8.in#L163
+STOPSIGNAL SIGQUIT
+
+EXPOSE 9000
+"""
+    else:
+        extra_pkgs = []
+        extra_env = {}
+        cmd = ["php", "-a"]
+        custom_end = common_end
+
+    return LanguageStackContainer(
+        name=str(php_variant),
+        version=php_version,
+        pretty_name=f"{str(php_variant).upper()} {php_version}",
+        package_name=f"{php_variant}{php_version}-image",
+        os_version=os_version,
+        package_list=[
+            f"php{php_version}-cli",
+            "php-composer",
+            f"php{php_version}-curl",
+            f"php{php_version}-zip",
+            f"php{php_version}-zlib",
+            f"php{php_version}-phar",
+            f"php{php_version}-mbstring",
+        ]
+        + extra_pkgs,
+        replacements_via_service=[
+            Replacement("%%composer_version%%", package_name="php-composer2")
+        ],
+        cmd=cmd,
+        entrypoint=["docker-php-entrypoint"],
+        env={
+            "PHP_VERSION": php_version,
+            "PHP_INI_DIR": f"/etc/php{php_version}/",
+            "PHPIZE_DEPS": f"php{php_version}-devel autoconf gawk gcc make file",
+            "COMPOSER_VERSION": "%%composer_version%%",
+            **extra_env,
+        },
+        extra_files={
+            "/usr/local/bin/docker-php-entrypoint": _php_entrypoint(php_variant),
+            "/usr/local/bin/docker-php-source": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-configure": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-enable": _EMPTY_SCRIPT,
+            "/usr/local/bin/docker-php-ext-install": f"""#!/bin/sh
+set -eox pipefail
+zypper -n in --no-recommends php{php_version}-$1
```

Comment:
please install with recommends as that adds a couple of package-specified extra extensions